src/FedoraApi.php

```diff
@@ -305,4 +307,79 @@ public function getGraph(ResponseInterface $response)
         }
         return $graph;
     }
+
+    /**
+     * Creates version in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param string $timestamp Timestamp for Memento version
+     * @param string $content String or binary content
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function createVersion(
+        $uri = '',
+        $timestamp = '',
+        $content = null,
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false];
+        if ($timestamp != ''){
+            $headers['Memento-Datetime'] = $timestamp;
+        }
+        if ($content != null){
+            $options['body'] = $content;
+        }
+        $options['headers'] = $headers;
+
+        return $this->client->request(
+            'POST',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Gets list of versions in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function getVersions(
+        $uri = '',
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false, 'headers' => $headers];
+
+        return $this->client->request(
+            'GET',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Helper method to get the Headers for a resource
+     * and parse the timemap header from it
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return string
+     */
+    private function getTimemapURI($uri, $headers){
+        $resource_headers = $this->getResourceHeaders($uri, $headers);
+        $parsed_link_headers = Psr7\parse_header($resource_headers->getHeader('Link'));
+        $timemap_uri = NULL;
+        foreach($parsed_link_headers as $link_header){
+            if (isset($link_header['rel']) && $link_header['rel'] == "timemap") {
```

Comment:
Instead of the `foreach` and this `if` you can search the array like this [`$match = array_search('describes', array_column($link_header, 'rel'));`](https://github.com/Islandora-CLAW/Crayfish/blob/dev/Recast/src/Controller/RecastController.php#L250).

The `array_column()` command returns an array of the `rel` elements from `$link_headers`. Then `array_search` searchs for your value.

Yours (I think) could look like
```
$timemap_index = array_search('timemap', array_column($parsed_link_headers, 'rel'));
if (is_int($timemap_index)) {
    $timemap_uri = $parsed_link_headers[$timemap_index][0];
}
```
Perhaps the fact that the Psr7\parse_header function returns an array with both named and unnamed elements is a problem, but it might be worth a try.

Reply:
I will test this out.

Reply:
works well: b0661d4